### PR TITLE
Bug fixes

### PR DESCRIPTION
**Pull request by Shilpi75**
1. Minor changes in Settings Activity strings and Ui
2. Vibration bug (Oreo, Samsung) solved
3. Reminder range validation
4. Repeat selection of days on task edit
5. Toast on app on/off
6. Motorola bugs fixed
7. Sound before voice alarms

### CODE REVIEW

File: app/src/main/java/app/tasknearby/yashcreations/com/tasknearby/utils/AppUtils.java [hunk 1]
```diff
@@ -223,4 +226,17 @@ private static String getWeekdayNameById(int index) {
                 return "";
         }
     }
+
+    public static boolean validateReminderRange(Context context, String input) {
```
[the remark this answers is not in the document]
Comment (Shilpi75) [reply app/src/main/java/app/tasknearby/yashcreations/com/tasknearby/utils/AppUtils.java]: Yes. Done

File: app/src/main/java/app/tasknearby/yashcreations/com/tasknearby/AlarmActivity.java [hunk 2]
```diff
@@ -243,9 +243,11 @@ public void onMapReady(GoogleMap googleMap) {
     protected void onStart() {
         super.onStart();
         if (isVoiceReminderEnabled) {
-            mVoiceAlarmRinger.startSpeaking();
+            mVoiceAlarmRinger.startVoiceAlarms();
         } else {
-            mAlarmVibrator.startVibrating();
+            if (mAlarmVibrator.mPattern.length != 0) {
+                mAlarmVibrator.startVibrating();
+            }
```
[the remark this answers is not in the document]
Comment (Shilpi75) [reply app/src/main/java/app/tasknearby/yashcreations/com/tasknearby/AlarmActivity.java]: Got it. Done

File: app/src/main/java/app/tasknearby/yashcreations/com/tasknearby/MainActivity.java [hunk 3]
```diff
@@ -66,6 +66,8 @@
 
     private FirebaseAnalytics mFirebaseAnalytics;
 
+    private boolean isAppIntiated;
```
[the remark this answers is not in the document]
Comment (Shilpi75) [reply app/src/main/java/app/tasknearby/yashcreations/com/tasknearby/MainActivity.java]: Better. Done

File: app/src/main/java/app/tasknearby/yashcreations/com/tasknearby/MainActivity.java [hunk 4]
```diff
@@ -115,6 +117,7 @@ protected void onCreate(Bundle savedInstanceState) {
 
     @Override
     protected void onStart() {
+        isAppIntiated = false;
```
[the remark this answers is not in the document]
Comment (Shilpi75) [reply app/src/main/java/app/tasknearby/yashcreations/com/tasknearby/MainActivity.java]: Done

File: app/src/main/res/values/strings.xml [hunk 5]
```diff
@@ -75,7 +75,7 @@
     <string name="creator_location_name_hint">Location Name</string>
     <string name="creator_reminder_range">Reminder range :</string>
     <string name="creator_reminder_range_default" translatable="false">75</string>
-    <string name="creator_alarm_title">Alarm</string>
+    <string name="creator_alarm_title">Alarm Tone</string>
```
[the remark this answers is not in the document]
Comment (Shilpi75) [reply app/src/main/res/values/strings.xml]: My bad. Done

File: app/src/main/java/app/tasknearby/yashcreations/com/tasknearby/utils/alarm/VoiceAlarmRinger.java [hunk 6]
```diff
@@ -87,4 +113,25 @@ private void showSnackbar() {
...
+    private void stopSound() {
+        if (mediaPlayer != null && mediaPlayer.isPlaying()) {
+            mediaPlayer.stop();
+        }
+    }
+
+    public void startVoiceAlarms() {
+        startSound();
+        startSpeaking();
+    }
```
[the remark this answers is not in the document]
Comment (Shilpi75) [reply app/src/main/java/app/tasknearby/yashcreations/com/tasknearby/utils/alarm/VoiceAlarmRinger.java]: Done

File: app/src/main/java/app/tasknearby/yashcreations/com/tasknearby/utils/alarm/AlarmVibrator.java [hunk 7]
```diff
@@ -47,6 +47,9 @@ public AlarmVibrator(Context context) {
      * Starts the vibration as per the user's preferences.
      */
     public void startVibrating() {
+        // If the settings is 'Don't Vibrate'.
+        if (mPattern.length == 0)
```
Comment (vermayash8) [inline app/src/main/java/app/tasknearby/yashcreations/com/tasknearby/utils/alarm/AlarmVibrator.java]: If your `if` statement has the `then` part in a new line, please surround it with braces. You can ignore the braces if the `if` contains a single statement on the same line. This issue came twice in this pull request.

**Pull request merged**